stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/feature/messages/list/internal/PollResultsDialogFragment.kt

```diff
@@ -0,0 +1,167 @@
+ */
+
+package io.getstream.chat.android.ui.feature.messages.list.internal
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.appcompat.app.AppCompatDialogFragment
+import androidx.appcompat.widget.Toolbar
+import androidx.core.content.ContextCompat
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.RecyclerView
+import io.getstream.chat.android.models.Poll
+import io.getstream.chat.android.models.Vote
+import io.getstream.chat.android.ui.ChatUI
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.databinding.StreamUiFragmentPollResultsBinding
+import io.getstream.chat.android.ui.databinding.StreamUiItemResultBinding
+import io.getstream.chat.android.ui.databinding.StreamUiItemResultUserBinding
+import io.getstream.chat.android.ui.utils.extensions.streamThemeInflater
+
+/**
+ * Represent the bottom sheet dialog that allows users to pick attachments.
+ */
+public class PollResultsDialogFragment : AppCompatDialogFragment() {
+
+    private var _binding: StreamUiFragmentPollResultsBinding? = null
+    private val binding get() = _binding!!
+    private lateinit var poll: Poll
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        _binding = StreamUiFragmentPollResultsBinding.inflate(requireContext().streamThemeInflater, container, false)
+        return binding.root
+    }
+
+    override fun getTheme(): Int = R.style.StreamUiBottomSheetDialogTheme
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupDialog()
+    }
+
+    /**
+     * Initializes the dialog.
+     */
+    private fun setupDialog() {
+        setupToolbar(binding.toolbar)
+        binding.question.text = poll.name
+        binding.optionList.adapter = ResultsAdapter(poll)
+    }
+
+    private fun setupToolbar(toolbar: Toolbar) {
+        toolbar.setNavigationOnClickListener { dismiss() }
+        ContextCompat.getDrawable(requireContext(), R.drawable.stream_ui_arrow_left)?.apply {
+            setTint(ContextCompat.getColor(requireContext(), R.color.stream_ui_black))
+        }?.let(toolbar::setNavigationIcon)
+        toolbar.setTitle(getString(R.string.stream_ui_poll_results_title))
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    public companion object {
+        public const val TAG: String = "create_poll_dialog_fragment"
+
+        /**
+         * Creates a new instance of [PollResultsDialogFragment].
+         *
+         * @return A new instance of [PollResultsDialogFragment].
+         */
+        public fun newInstance(poll: Poll): PollResultsDialogFragment = PollResultsDialogFragment().apply {
+            this.poll = poll
+        }
+    }
+
+    private class ResultsAdapter(
```

Comment:
Question: Is there a reason why the `adapter` classes are internal to the `Fragment` and the `holder` classes are in turn private in the `adapter`? We have packages for view-holder classes.

Reply:
Lets keep them in for now.